tfe/data_source_version_files.go

```diff
@@ -0,0 +1,67 @@
+package tfe
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"log"
+	"os"
+
+	slug "github.com/hashicorp/go-slug"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceTFEVersionFiles() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceTFEVersionFilesRead,
+
+		Schema: map[string]*schema.Schema{
+			"source_path": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"checksum": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceTFEVersionFilesRead(d *schema.ResourceData, meta interface{}) error {
+	sourcePath := d.Get("source_path").(string)
+
+	log.Printf("[DEBUG] Hashing the source path files: %s", sourcePath)
+	chksum, err := hashPolicies(sourcePath)
+	if err != nil {
+		return fmt.Errorf("Error generating the checksum for the source path files: %v", err)
+	}
+	d.SetId(chksum)
+	d.Set("checksum", chksum)
+
+	return nil
+}
+
+func hashPolicies(path string) (string, error) {
```

Comment:
Nitpick: should this be something like `hashDirectory`? I could see us using this data source for other things down the road (for instance, adding a `RegistryModuleVersion` for registry modules that aren't connected to VCS), so the given directory might not always contain policies!